mlflow/server/js/src/model-registry/components/ModelVersionView.js

```diff
@@ -52,20 +54,72 @@ export class ModelVersionViewImpl extends React.Component {
   componentDidMount() {
     const pageTitle = `${this.props.modelName} v${this.props.modelVersion.version} - MLflow Model`;
     Utils.updatePageTitle(pageTitle);
+    this.checkDeploymentStatus(this.props.modelVersion.run_id)
   }
 
-  handleModelDeployment = () => {
-    if (this.state.isModelDeployed == true){
-       this.setState({
-        isModelDeployed: false,
-        modelInferenceAPI:'Not Deployed'
+  checkDeploymentStatus(modelId) {
+    var xhr = new XMLHttpRequest()
+    xhr.addEventListener('load', () => {
+      if (xhr.responseText === "deployed") {
+        this.setState({
+          isModelDeployed: true,
+          modelInferenceAPI: MODEL_INFERENCE_API + '/' + this.props.modelVersion.run_id,
+        });
+      }
+      else {
+        this.setState({
+          isModelDeployed: false,
+          modelInferenceAPI: "Not Deployed",
+        });
+      }
     });
+    xhr.onerror = function (e) {
+      console.log(e)
+      this.setState({
+        isModelDeployed: false,
+        modelInferenceAPI: 'Server not reachable.'
+      });
+    }.bind(this);
+    xhr.open('GET', MODEL_DEPLOYMENT_URL + "?runId=" + modelId)
+    xhr.send()
```

Comment:
Refactor this to another file. We want to keep logic in components, specially API calls logic as low as possible.

Reply:
Done